SWIPs/swip-multiple_payment_processing_support.md

```diff
@@ -27,58 +27,177 @@ This SWIP is part of a series of SWIPs (but can be implemented on its own). To s
+	| ---------- | --------------- |
+	| Currency to use | Wei (Ether)                    |
+	| Price oracle    | HonMon oracle <SWIP reference> |
+	| Payment method  | Chequebook contract            |
+
+4. Compute the weighted preference of each remaining triplet from node A and node B as:
+
+	```golang
+	weightedPreference := normalized(dimensions.currency) * normalized(triplet[n].weight) + 
+		+ normalized(dimensions.provider) * normalized(triplet[n].provider.weight) + 
+		+ normalized(dimensions.oracle) * normalized(triplet[n].oracle.weight)
+	```
+
+	Since weights and dimensions do not need to add up to a particular value (e.g. 100) they need to be normalized.
+	As an example, let's take the first generated triplet (`[rif, lumino, rifOracleA]`). The normalized dimensions and weights are:
+	
+	***Sum of all dimension values = 100***
+	* Normalized currency dimension: 70 / 100 = 0.7
+	* Normalized provider dimension: 20 / 100 = 0.2
+	* Normalized oracle dimension: 10 / 100 = 0.1
+
+	***Sum of all currency weights = 90***
+	* RIF normalized weight = 36 / 90 = 0.4
+
+	***Sum of all provider weights for RIF = 60***
+	* Lumino normalized weight = 15 / 60 = 0.25
+
+	***Sum of all oracle weights for RIF = 140***
+	* oracleB normalized weight = 70 / 140 = 0.5
 
-Nodes must be able to reach an agreement in any dimension of the preference list, or the fallback option for each dimension of the preference list will be chosen:
+	Then the ```weightedPreference``` for this triplet is:
 
-| Dimension  | Fallback option |
-| ---------- | --------------- |
-| Currency to use | Wei (Ether)                    |
-| Price oracle    | HonMon oracle <SWIP reference> |
-| Payment method  | Chequebook contract            |
+	```golang
+    weightedPreference := [0.7 * 0.4 + 0.2 * 0.25 + 0.1 * 0.5]
+	weightedPreference := 0.38
+	```
 
-Please refer to the picture below to see how preferences are resolved during the handshake:
+5. Add the weights of matching triplets from A and B and choose the one with the highest cumulative preference value:
 
-![Handshake.svg](./../assets/multiple-payment_processing_support/Handshake.svg)
+	```golang
+	highestPreference := 0
+	combinedWeightedPreference := 0
+	
+	for i, weightedPreferenceA := range weightedPreferencesA {
+		combinedWeightedPreference = weightedPreferenceA + weightedPreferencesB[i]
+		if highestPreference < combinedWeightedPreference {
+			highestPreference = combinedWeightedPreference
+		}
+	}
+	```
+
+6. If there is more than one triplet with the same ```weightedPreference``` value, both nodes will compute for each colliding triplet a tie breaker as follows:
+
+	```golang
+	tieBreaker := hash(triplet.coin || triplet.provider || triplet.oracle)
+	```
+	
+	and then each node will select the triplet with the minimum ```tieBreaker``` value.
+
+The oracle negotiated during this process will override the default oracle described in [swip-honey_to_money](./swip-honey_to_money.md).
 
 ### Technical details
```

Comment:
Consider changing the heading from "technical details" to exploration of implementation. 
Maybe we should put this all in the implementation section, actually?